projects/core/src/organization/facade/org-unit.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Observable, queueScheduler } from 'rxjs';
+import { filter, map, observeOn, take, tap } from 'rxjs/operators';
+import { StateWithProcess } from '../../process/store/process-state';
+import { LoaderState } from '../../state/utils/loader/loader-state';
+import { AuthService } from '../../auth/facade/auth.service';
+import { StateWithOrganization } from '../store/organization-state';
+import { OrgUnitActions } from '../store/actions/index';
+import {
+  getOrgUnitState,
+  getOrgUnitList,
+} from '../store/selectors/org-unit.selector';
+import { B2BUnitNode, B2BUnitNodeList } from '../../model';
+
+@Injectable()
+export class OrgUnitService {
+  private user$ = this.authService.getOccUserId();
+
+  constructor(
+    protected store: Store<StateWithOrganization | StateWithProcess<void>>,
+    protected authService: AuthService
+  ) {}
+
+  private loadOrgUnit(orgUnitId: string) {
+    this.user$
+      .pipe(take(1))
+      .subscribe(userId =>
+        this.store.dispatch(
+          new OrgUnitActions.LoadOrgUnit({ userId, orgUnitId })
+        )
+      );
+  }
+
+  loadOrgUnits() {
+    this.user$
+      .pipe(take(1))
+      .subscribe(userId =>
+        this.store.dispatch(new OrgUnitActions.LoadOrgUnits({ userId }))
+      );
+  }
+
+  private getOrgUnitState(orgUnitId: string) {
+    return this.store.select(getOrgUnitState(orgUnitId));
+  }
+
+  private getOrgUnitsList(): Observable<LoaderState<B2BUnitNodeList>> {
+    return this.store.select(getOrgUnitList());
+  }
+
+  get(orgUnitId: string): Observable<B2BUnitNode> {
+    return this.getOrgUnitState(orgUnitId).pipe(
+      observeOn(queueScheduler),
+      tap(state => {
+        if (!(state.loading || state.success || state.error)) {
```

Comment:
You should make such side effects inside tap. Tap is perfect for logging or debugging. If you want to redirect stream use `switchMap`;